frameworks/cassandra/universe/config.json

```diff
@@ -6,8 +6,9 @@
       "description": "DC/OS Apache Cassandra service configuration properties",
       "properties": {
         "name": {
-          "description": "The name of the Cassandra service instance.",
+          "description": "Unique name for the service instance consisting of a series of words separated by slashes. Each word must be at least 1 alphanumeric character and may only contain digits (`0-9`), dashes (`-`), dots (`.`), and lowercase letters (`a-z`). The word may not begin or end with a dash",
           "type": "string",
+          "pattern": "^(\\/?((\\.\\.)|(([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9]))?($|\\/))+$",
```

Comment:
A few questions:
- do we really need to escape forward slashes?
- do we need to escape a dash in a character class if it's the last character? All regex engines I've seen so far do not require this.

Reply:
Thanks @porridge for the review. This regex has been tested manually during HDFS release, by creating a cluster and giving it an incorrect name. But I think it makes sense to have a unit test for this. 
As for the engine, locally I've tested it in Python and using some free tools which uses a Javascript regex engine.
Finally, I'll rename the PR and make it pertinent to the specific issue we're addressing here.